Order output for Get*Months Get*Days

huehnerhose opened the pull request:
I have some use cases where I do not have an actual time.Time, but only need the list of months in a given locale. `GetLongMonths` etc. seem to want to give me exactly that, but they aren't ordered. 

I see that as a bug, hence this PR. 

As far as I see only the USA, Canada and Japan see sunday as first day of the week. So I think it is sufficient to handle their speciality inside the getter function rather than introducing a new order variable in all locale files.

goodsign commented:
Hey @huehnerhose 

I think this makes total sense, thank you for the contribution! Could you please add a couple of simple tests which will cover the locale-based order (Sun first / Mon first)?

Thank you!

huehnerhose commented:
Hi @goodsign 

IMHO I added reasonable tests.
I added a new locale EnGB, as copy of of EnUS, since GB uses monday as first day week and we need to differentiate those cases. Although there are many other english-speaking countries, I think it is sufficient to have those two "big names" as proxies.

goodsign commented:
Hey @huehnerhose 

Yeah, that's definitely reasonable and sufficient. Thank you very much for contributing!
